cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/Filters/FilterCollection/FilterConditionInput/index.tsx

```diff
@@ -0,0 +1,150 @@
+import If from 'components/If';
+import { OPERATOR_VALUES } from 'components/PluginJSONCreator/constants';
+import PluginInput from 'components/PluginJSONCreator/Create/Content/PluginInput';
+import { ICreateContext, IWidgetInfo } from 'components/PluginJSONCreator/CreateContextConnect';
+import * as React from 'react';
+
+const styles = (): StyleRules => {
+  return {
+    filterConditionInput: {
+      '& > *': {
+        marginTop: '10px',
+        marginBottom: '10px',
+      },
+    },
+  };
+};
+
+export enum FilterConditionMode {
+  Operator = 'operator',
+  Expression = 'expression',
+}
+
+interface IFilterConditionInputProps extends WithStyles<typeof styles>, ICreateContext {
+  filterID: string;
+}
+
+const FilterConditionInputview: React.FC<IFilterConditionInputProps> = ({
+  classes,
+  filterID,
+  filterToCondition,
+  setFilterToCondition,
+  widgetInfo,
+}) => {
+  const existingCondition = filterToCondition[filterID];
+
+  const [conditionMode, setConditionMode] = React.useState(
+    existingCondition.expression ? FilterConditionMode.Expression : FilterConditionMode.Operator
+  );
+
+  const allWidgetNames = widgetInfo
+    ? Object.values(widgetInfo)
+        .map((info: IWidgetInfo) => info.name)
+        .filter((widgetName) => widgetName !== undefined && widgetName !== null)
+    : [];
+
+  React.useEffect(() => {
+    // reset condition data
+    if (conditionMode === FilterConditionMode.Expression) {
+      setFilterToCondition((prevObjs) => ({
+        ...prevObjs,
+        [filterID]: { expression: existingCondition.expression },
+      }));
+    } else {
+      setFilterToCondition((prevObjs) => ({
+        ...prevObjs,
+        [filterID]: {
+          property: existingCondition.property,
+          operator: existingCondition.operator,
+          value: existingCondition.value,
+        },
+      }));
+    }
+  }, [conditionMode]);
+
+  function setFilterCondition(conditionProperty: string) {
+    return (val) => {
+      setFilterToCondition((prevObjs) => ({
+        ...prevObjs,
+        [filterID]: { ...prevObjs[filterID], [conditionProperty]: val },
+      }));
+    };
+  }
+
+  return (
+    <If condition={existingCondition}>
+      <div className={classes.filterConditionInput}>
+        <Heading type={HeadingTypes.h6} label="Show these widgets by the following condition..." />
+        <PluginInput
+          widgetType={'radio-group'}
+          value={conditionMode}
+          onChange={setConditionMode}
+          label={'Condition Type'}
+          options={Object.values(FilterConditionMode).map((mode) => ({ id: mode, label: mode }))}
+          layout={'inline'}
+        />
+        <If condition={conditionMode === FilterConditionMode.Expression}>
+          <PluginInput
+            widgetType={'textbox'}
+            value={existingCondition.expression ? existingCondition.expression : ''}
+            onChange={setFilterCondition('expression')}
```

Comment:
If there is no enum for `expression` and ` condition` then we should probably introduce it. Loosely using strings for types might introduce problems if we miss consistency.

Reply:
Resolved